src/bundle.c

```diff
@@ -248,7 +248,7 @@ static gboolean input_stream_read_uint64_all(GInputStream *stream,
 	return res;
 }
 
-#define SQUASHFS_MAGIC			0x73717368
+#define SQUASHFS_MAGIC			GUINT32_FROM_LE(0x73717368)
```

Comment:
This should be `GUINT32_TO_LE` because you are using it to compare with a little endian `GUINT32`. Doesn't matter except for PDP Endian, but makes he intent clearer

Reply:
Yes, @a3f, you're right. @ukleinek to you want to swap it or should I?

Reply:
I thought a bit about that and I think `GUINT32_FROM_LE` is the right one. We're reading a 32bit little endian number from the image and interpret it as native number. So that's what should be done with the little endian constant that we're expecting to read, don't we?

Alternatively we could compare byte by byte.

Reply:
If we add some types it becomes clearer:

```
__le32 guint32_from_image;
__host32 guint32_integer_literal;
__host32 GUINT32_FROM_LE(__le32);
__le32 GUINT32_TO_LE(__host32);
```

So valid combinations are either `if (GUINT32_FROM_LE(guint32_from_image) == guint32_integer_literal)` or `if (GUINT32_TO_LE(guint32_integer_literal) == guin32_from_image)`.

Reply:
ok, convinced